base_tier_validation/models/tier_validation.py

```diff
@@ -271,9 +271,13 @@ def write(self, vals):
             ):
                 raise ValidationError(_("The operation is under validation."))
         if vals.get(self._state_field) in (self._state_from + [self._cancel_state]):
```

Comment:
_state_from already support more than one state, see [in purchases](https://github.com/OCA/purchase-workflow/blob/14.0/purchase_tier_validation/models/purchase_order.py#L10) 

I think the problem is in this condition, which needs to be a bit more complex an check that the origin state is not in self._state_from. In other words if you go from one of the _state_from to the same or other of the _state_from, the review shouln't be unlinked.